packages/office-ui-fabric-react/src/components/ProgressIndicator/ProgressIndicator.styles.ts

```diff
@@ -1,41 +1,131 @@
-import { IProgressIndicatorStyleProps, IProgressIndicatorStyles } from './ProgressIndicator.types';
+import {
+  FontSizes,
+  FontWeights,
+  HighContrastSelector,
+  keyframes,
+  noWrap,
+} from '../../Styling';
+import { getRTL } from '../../Utilities';
+import {
+  IProgressIndicatorStyleProps,
+  IProgressIndicatorStyles,
+} from './ProgressIndicator.types';
 
 export const getStyles = (
   props: IProgressIndicatorStyleProps
 ): IProgressIndicatorStyles => {
-  const { className } = props;
+  const isRTL = getRTL();
+  const {
+    className,
+    indeterminate,
+    theme,
+    barHeight = 2,
+  } = props;
+
+  const { palette, semanticColors } = theme;
+
+  const marginBetweenText = 8;
+  const textHeight = 18;
+
+  const indeterminateProgress = keyframes({
+    '0%': {
+      left: '-30%',
+    },
+    '100%': {
+      left: '100%',
+    }
+  });
+  const indeterminateProgressRTL = keyframes({
+    '100%': {
+      right: '-30%',
+    },
+    '0%': {
+      right: '100%',
+    }
+  });
 
   return ({
     root: [
       'ms-ProgressIndicator',
-      {},
+      {
+        fontWeight: FontWeights.regular,
+      },
       className
     ],
 
     itemName: [
       'ms-ProgressIndicator-itemName',
-      {}
+      noWrap,
+      {
+        color: semanticColors.bodyText,
+        fontSize: FontSizes.medium,
+        paddingTop: `${marginBetweenText / 2}px`,
+        lineHeight: `${textHeight + 2}px`,
+      }
     ],
 
     itemDescription: [
       'ms-ProgressIndicator-itemDescription',
-      {}
+      {
+        color: semanticColors.bodySubtext,
+        fontSize: FontSizes.xSmall,
+        lineHeight: textHeight,
+      }
     ],
 
     itemProgress: [
       'ms-ProgressIndicator-itemProgress',
-      {}
+      {
+        position: 'relative',
+        overflow: 'hidden',
+        height: `${barHeight}px`,
+        padding: `${marginBetweenText}px 0`,
+      }
     ],
 
     progressTrack: [
       'ms-ProgressIndicator-progressTrack',
-      {}
+      {
+        position: 'absolute',
+        width: '100%',
+        height: `${barHeight}px`,
+        backgroundColor: palette.neutralLight,
+
+        selectors: {
+          [HighContrastSelector]: {
+            borderBottom: '1px solid WindowText',
+          }
+        }
+      }
     ],
 
     progressBar: [
       'ms-ProgressIndicator-progressBar',
-      {},
+      {
+        backgroundColor: palette.themePrimary,
+        height: `${barHeight}px`,
+        position: 'absolute',
+        transition: 'width .3s ease',
+        width: 0,
+
+        selectors: {
+          [HighContrastSelector]: {
+            backgroundColor: 'WindowText',
+          }
+        }
+      },
+      !indeterminate && {
+        transition: 'width .15s linear'
+      },
+      indeterminate && [
+        {
+          position: 'absolute',
+          minWidth: '33%',
+          background: `linear-gradient(to right, transparent 0%, ${palette.themePrimary} 50%, transparent 100%)`,
```

Comment:
this may not work in RTL. I'm not sure we will flip "to right" in the linear-gradient... Test in RTL, workaround is (getRTL() ? 'left': 'right')

Reply:
The gradient is symmetrical so it doesn't *currently* make a difference between to right or to left, but if it were to change it'd be nice to flip.

Reply:
It does flip.
![image](https://user-images.githubusercontent.com/7110658/39274931-68a28bbc-4898-11e8-9a10-a1a9bf9e5eed.png)